src/opera_disp_tms/generate_coh_tiles.py

```diff
@@ -0,0 +1,179 @@
+        path = '/'.join(s3_path.split('/')[3:])
+        try:
+            S3.download_file(bucket, path, s3_path.split('/')[-1])
+            print(f'Downloaded {s3_path}')
+        except ClientError:
+            print(f'{s3_path} not found')
+            return
+        return s3_path
+    else:
+        return s3_path
+
+
+def create_coh_tile(bbox: Iterable[int], coh_prod: str = 'summer_vv_COH12', gtiff: bool = True) -> str:
+    """Create a mosaic of coherence tiles for a given bounding box
+
+    Args:
+        bbox: bounding box [min_lon, min_lat, max_lon, max_lat]
+        coh_prod: coherence product name
+        gtiff: save as GeoTIFF or COG
+
+    Returns:
+        str: path to the mosaic
+    """
+    min_lon, min_lat, max_lon, max_lat = bbox
+    s3_paths = []
+    for lon in range(min_lon, max_lon):
+        for lat in range(min_lat, max_lat):
+            s3_paths.append(create_coh_s3_path(lon, lat))
+
+    s3_paths = [create_coh_s3_path(lon, lat) for lon, lat in product(range(min_lon, max_lon), range(min_lat, max_lat))]
+
+    s3_paths = [download_coh(x) for x in s3_paths]
+    names = [x.split('/')[-1] for x in s3_paths if x is not None]
+
+    min_lat_str = f'N{abs(min_lat):02d}' if min_lat >= 0 else f'S{abs(min_lat):02d}'
+    min_lon_str = f'E{abs(min_lon):03d}' if min_lon >= 0 else f'W{abs(min_lon):03d}'
+    max_lat_str = f'N{abs(max_lat):02d}' if max_lat >= 0 else f'S{abs(max_lat):02d}'
+    max_lon_str = f'E{abs(max_lon):03d}' if max_lon >= 0 else f'W{abs(max_lon):03d}'
+    bbox_str = f'{min_lat_str}{min_lon_str}_{max_lat_str}{max_lon_str}'
+
+    product_name = f'{coh_prod}_{bbox_str}'
+    vrt_path = f'{product_name}.vrt'
+    output_path = f'{product_name}.tif'
+
+    if len(names) > 0:
+        print('Building mosaic...')
+        vrt = gdal.BuildVRT(vrt_path, names)
+        vrt.FlushCache()
+        vrt = None
+
+        if gtiff:
+            opts = ['TILED=YES', 'COMPRESS=LZW', 'NUM_THREADS=ALL_CPUS']
+            warp_options = gdal.WarpOptions(format='GTiff', dstSRS='EPSG:3857', xRes=30, yRes=30, creationOptions=opts)
+        else:
+            warp_options = gdal.WarpOptions(format='COG', dstSRS='EPSG:3857', xRes=30, yRes=30)
+        gdal.Warp(output_path, vrt_path, options=warp_options)
+
+        Path(vrt_path).unlink()
+        for name in names:
+            Path(name).unlink()
+        print('Mosaic done')
+    else:
+        print('No tiles found')
+
+    return output_path
+
+
+def split_range(start_value: int, end_value: int, n: int) -> list:
+    """Split a range into n parts
+    Args:
+        start_value: start of the range
+        end_value: end of the range
+        n: number of parts
+
+    Returns:
+        list of tuples with the ranges
+    """
+    step = (end_value - start_value) // n
+    ranges = []
+    for i in range(n):
+        start = start_value + i * step
+        end = start_value + (i + 1) * step
+        ranges.append((start, end))
+    return ranges
```

Comment:
Does it matter that we are losing the remainder here? For example, `split_range(0, 11, 2)` will return `[(0, 5), (5, 10)]` rather than `[(0, 5), (5, 11)]`. Something like the following is a bit simpler and allows for the range to keep the remainder at the end.
```suggestion
def split_range(start_value: int, end_value: int, n: int) -> list:
    """Split a range into n parts
    Args:
        start_value: start of the range
        end_value: end of the range
        n: number of parts

    Returns:
        list of tuples with the ranges
    """
    step = (stop-start) // n
    offset = (stop-start) % n
    ranges = [(a, b) for (a, b) in zip(range(start, stop, step), range(start+step, stop-offset+step, step))]
    ranges[-1] = (ranges[-1][0], ranges[-1][1]+offset)
    return ranges
```

Reply:
Also, I supposed neither of these handle the anti-meridian, but as long as the tile sizes divide 180 evenly that shouldn't be an issue.

Reply:
Ack, yes losing the remainder is an issue. Thanks for the find/fix.